repose-aggregator/components/filters/keystone-v2-filter/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -116,25 +116,17 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
       lazy val requestHandler = new KeystoneRequestHandler(keystoneV2Config.getIdentityService.getUri, akkaServiceClient, traceId)
       lazy val isSelfValidating = Option(config.getIdentityService.getUsername).isEmpty ||
         Option(config.getIdentityService.getPassword).isEmpty
-      lazy val tenantFromUriOption: Option[String] = {
-        def extractTenantIdFromUri(uri: String, regexes: Seq[Regex]): Option[String] = {
-          if (regexes.isEmpty) {
-            throw UnparseableTenantException("Could not parse tenant from the URI")
-          } else {
-            val regex = regexes.head
-            uri match {
-              case regex(tid, _*) => Some(tid)
-              case _ => extractTenantIdFromUri(uri, regexes.tail)
-            }
-          }
-        }
-
-        Option(config.getTenantHandling.getValidateTenant) flatMap { validateTenantConfig =>
-          Option(validateTenantConfig.getUriExtractionRegex) flatMap { uriExtractionRegexList =>
-            extractTenantIdFromUri(request.getRequestURI, uriExtractionRegexList.asScala.map(_.r))
-          }
-        }
-      }
+      lazy val tenantFromUriOption: Option[String] =
+        Option(config.getTenantHandling.getValidateTenant).flatMap({ validateTenantConfig =>
+          Option(validateTenantConfig.getUriExtractionRegex).flatMap({ uriExtractionRegexList =>
+            uriExtractionRegexList.asScala.toStream.map(_.r).flatMap({ uriExtractionRegex: Regex =>
+                          request.getRequestURI match {
+                            case uriExtractionRegex(tenantId, _*) => Option(tenantId)
+                            case _ => None
+                          }
+                        }).headOption
```

Comment:
Most of that was due to the fallout of fighting through the required syntax to make it all work and we knew the dot notation would definitely work. Then the innermost one actually did require the doubling of the delimiters. So the fallout was they all had them. I just pushed the removal of the doubling from the outer two and the whitespace on the match statement.